health-checker/src/main/java/io/stargate/health/metrics/impl/PrefixingMetricRegistry.java

```diff
@@ -0,0 +1,349 @@
+package io.stargate.health.metrics.impl;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+import java.util.SortedMap;
+import java.util.SortedSet;
+import java.util.TreeMap;
+import java.util.TreeSet;
+import java.util.concurrent.ConcurrentMap;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.MetricFilter;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.MetricRegistryListener;
+import com.codahale.metrics.MetricSet;
+import com.codahale.metrics.Timer;
+
+/**
+ * Wraps an existing registry to prefix every new metric with a given string on the fly.
+ *
+ * <p>For clients of this class, it looks like they're interacting with a regular registry, when
+ * they are in fact only seeing one part of a bigger registry, that has other metrics under
+ * different prefixes.
+ *
+ * <p>Implementation note: unfortunately {@link MetricRegistry} is a class, not an interface. So we
+ */
+public class PrefixingMetricRegistry extends MetricRegistry {
```

Comment:
I'm not overjoyed with this, but it's a convenient way to quickly integrate with modules that already work with a metric registry. Adding the prefix in the client code requires significantly more work, DropWizard in particular does not offer a built-in way to do it.

If we don't scope the metrics, the different modules collide and an error is thrown.